tests/ecmp/test_ecmp_sai_value.py

```diff
@@ -0,0 +1,232 @@
+]
+
+offset_cmd = 'bcmcmd  "dump RTAG7_PORT_BASED_HASH 0 392 OFFSET_ECMP"'
+
+
+def parse_hash_seed(output, asic_name):
+    logger.info("Checking seed config: {}".format(output))
+    # RTAG7_HASH_SEED_A.ipipe0[1][0x16001500]=0: <HASH_SEED_A=0>
+    # Regular expression pattern to find both HASH_SEED_A and HASH_SEED_B
+    pattern = r'HASH_SEED_[A|B]=(0x?[0-9a-fA-F]?)'
+
+    matches = re.findall(pattern, output)
+    if len(matches) == 1:
+        logger.info("HASH_SEED value: {}".format(matches[0]))
+        numeric_value = matches[0]
+    else:
+        pytest.fail("Matched number of HASH_SEED is not correct.")
+    return numeric_value
+
+
+def parse_ecmp_offset(outputs):
+    # Regular expression pattern to extract OFFSET_ECMP values (hexadecimal)
+    pattern = r'OFFSET_ECMP=(0x?[0-9a-fA-F]?)'
+
+    # Extracted values
+    extracted_values = []
+
+    for line in outputs.splitlines():
+        line = line.strip()
+        matches = re.findall(pattern, line)
+        if len(matches) == 1:
+            value = matches[0]
+            extracted_values.append(value)
+        elif len(matches) == 0:
+            continue
+        else:
+            pytest.fail("Matched number of OFFSET_ECMP is not correct.")
+    return extracted_values
+
+
+def check_syncd_is_running(duthost):
+    """
+    Check if syncd container is running
+    """
+    output = duthost.shell("docker ps | grep syncd")["stdout"]
+    if "syncd" in output:
+        return True
+    else:
+        return False
+
+
+def check_hash_seed_value(duthost, asic_name, topo_type):
+    """
+    Check the value of HASH_SEED
+    t0: HASH_SEED is set to 0
+    t1: HASH_SEED is set to 0xa
+    """
+    pytest_assert(wait_until(300, 20, 0, check_syncd_is_running, duthost), "syncd is not running!")
+    if asic_name == "td2":
+        seed_cmd_input = seed_cmd_td2
+    elif asic_name == "td3":
+        seed_cmd_input = seed_cmd_td3
+    else:
+        seed_cmd_input = seed_cmd
+    for cmd in seed_cmd_input:
+        output = duthost.command(cmd, module_ignore_errors=True)["stdout_lines"][2].strip()
+        hash_seed = parse_hash_seed(output, asic_name)
+        if topo_type == "t1":
+            pytest_assert(hash_seed == '0xa', "HASH_SEED is not set to 0xa")
+        elif topo_type == "t0":
+            pytest_assert(hash_seed == '0', "HASH_SEED is not set to 0")
+
+
+def check_ecmp_offset_value(duthost, asic_name, topo_type):
+    """
+    Check the value of OFFSET_ECMP
+    TH/TH2: the count of 0xa is 67
+    TD2: the count of 0xa is 33
+    """
+    pytest_assert(wait_until(300, 20, 0, check_syncd_is_running, duthost), "syncd is not running!")
+    output = duthost.shell(offset_cmd, module_ignore_errors=True)['stdout']
+    offset_list = parse_ecmp_offset(output)
+    if topo_type == "t0":
+        offset_count = offset_list.count('0')
+        if asic_name == "td3":
+            # For TD3, RTAG7_PORT_BASED_HASH.ipipe0[1]: <OFFSET_ECMP=2,>
```

Comment:
for this one, could we raise a CSP to BRCM, what port it is and why it is 2?

Reply:
CS00012327297 is filed.